### DISCUSSION TITLE
Coverage of wrapping vtk classes table

### DISCUSSION THREAD

**Discussion by tkoyama010** [pyvista/pyvista#1719] [Ideas]
Is there a way to check the coverage of wrapping vtk classes?
If we have something like table. It will be a good information for contributors.
Inspired by 
https://kitware.github.io/vtk-examples/site/Python/Coverage/PythonVTKClassesNotUsed/
https://kitware.github.io/vtk-examples/site/Python/Coverage/PythonVTKClassesUsed/

**Comment by dcbr** [comment]
They seem to be using [this script](https://github.com/Kitware/vtk-examples/blob/master/src/Admin/VTKClassesUsedInExamples.py) to produce those tables. Using the method on line 116, you can obtain all vtk classes (fetched from the nightly docs). Then, to obtain all wrapped classes you could probably create a script that iteratively goes over all pyvista classes (using e.g. [inspect](https://docs.python.org/3/library/inspect.html)'s `getmembers`, `ismodule` and `isclass` methods) and check whether they subclass any vtk class, using for example:
```py
import pyvista
import numpy as np
cls = pyvista.Camera
is_vtk_subclass = np.any([base.__name__.startswith("vtk") for base in cls.__bases__])
```
This probably won't catch all wrappers (e.g. those not directly subclassing a vtk class), but should give a good first estimation.

**Reply by MatthewFlamm** [reply]
Might be easier to use all the objects imported in `pyvista._vtk`
